### ISSUE TITLE
Does pretraning script ONLY pretrain vision-language task?

### ISSUE THREAD

**Issue by xcvil** [OFA-Sys/OFA#376]
In 'data/pretrain_data/unify_dataset.py' line 472-475:

```python
if self.split == 'train' and self.dataset.data_cnt % 8 == 0:
        extra_samples += self.process_pure_text(0) if self.pure_text_dataset else []
        extra_samples += self.process_pure_image(0) if self.pure_image_dataset else []
        extra_samples += self.process_detection(0) if self.detection_dataset else []
```
why ```self.process_pure_text(0)``` 0 instead of ```index```?

**Comment by ZhangYuanhan-AI** [comment]
> In 'data/pretrain_data/unify_dataset.py' line 472-475:
> 
> ```python
> if self.split == 'train' and self.dataset.data_cnt % 8 == 0:
>         extra_samples += self.process_pure_text(0) if self.pure_text_dataset else []
>         extra_samples += self.process_pure_image(0) if self.pure_image_dataset else []
>         extra_samples += self.process_detection(0) if self.detection_dataset else []
> ```
> 
> why `self.process_pure_text(0)` 0 instead of `index`?

Yes, this part is really weird. I also need clarification about this. Do you have any thoughts?

**Comment by xcvil** [comment]
> > In 'data/pretrain_data/unify_dataset.py' line 472-475:
> > ```python
> > if self.split == 'train' and self.dataset.data_cnt % 8 == 0:
> >         extra_samples += self.process_pure_text(0) if self.pure_text_dataset else []
> >         extra_samples += self.process_pure_image(0) if self.pure_image_dataset else []
> >         extra_samples += self.process_detection(0) if self.detection_dataset else []
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > why `self.process_pure_text(0)` 0 instead of `index`?
> 
> Yes, this part is really weird. I also need clarification about this. Do you have any thoughts?

For language/detection/image_infilling task, OFA also uses `file_dataset`. And there, 

```python
def __getitem__(self, index):
        if self.data_cnt == self.row_count:
            print("reach the end of datafile, start a new reader")
            self.data_cnt = 0
            self._reader = self._get_reader()
        column_l = self._reader.readline().rstrip("\n").split(self.separator)
        self.data_cnt += 1
        column_l = [dtype(column_l[col_id]) for col_id, dtype in zip(self.selected_col_ids, self.dtypes)]
        return column_l
```

this part might explain why the index=0.

**Comment by ZhangYuanhan-AI** [comment]
Then it makes sense. Thank you!
